hadoop-hdds/rocks-native/src/main/java/org/apache/hadoop/hdds/utils/NativeLibraryLoader.java

```diff
@@ -132,15 +134,17 @@ private Optional<File> copyResourceFromJarToTemp(final String libraryName)
       }
 
       // create a temporary file to copy the library to
-      final File temp = File.createTempFile(libraryName, getLibOsSuffix());
+      final File temp = File.createTempFile(libraryName, getLibOsSuffix(),
+          new File(""));
       if (!temp.exists()) {
         return Optional.empty();
       } else {
         temp.deleteOnExit();
       }
 
       Files.copy(is, temp.toPath(), StandardCopyOption.REPLACE_EXISTING);
-      return Optional.ofNullable(temp);
+      ShutdownHookManager.get().addShutdownHook(temp::delete, 1);
+      return Optional.of(temp);
```

Comment:
`temp` could still be `null` if `File.createTempFile()` above doesn't have the permission to write to current dir right?

Reply:
it will throw an IOException.